numpy/lib/arraysetops.py

```diff
@@ -380,106 +381,130 @@ def setxor1d(ar1, ar2, assume_unique=False):
-        Default is False. ``np.in1d(a, b, invert=True)`` is equivalent
-        to (but is faster than) ``np.invert(in1d(a, b))``.
-
-        .. versionadded:: 1.8.0
+        If True, the values in the returned array are inverted (that is, False
+        where an element of `elements` is in `test_elements` and True
+        otherwise). Default is False. ``np.isin(a, b, invert=True)`` is
+        equivalent to (but is faster than) ``np.invert(isin(a, b))``.
 
     Returns
     -------
-    in1d : (M,) ndarray, bool
-        The values `ar1[in1d]` are in `ar2`.
+    isin : ndarray, bool
+        The values `elements[isin]` are in `test_elements`.
 
     See Also
     --------
+    in1d                  : The older version of this function that acts on
+                            flat arrays only.
     numpy.lib.arraysetops : Module with a number of other functions for
                             performing set operations on arrays.
 
     Notes
     -----
-    `in1d` can be considered as an element-wise function version of the
-    python keyword `in`, for 1-D sequences. ``in1d(a, b)`` is roughly
-    equivalent to ``np.array([item in b for item in a])``.
-    However, this idea fails if `ar2` is a set, or similar (non-sequence)
-    container:  As ``ar2`` is converted to an array, in those cases
-    ``asarray(ar2)`` is an object array rather than the expected array of
-    contained values.
+    `isin` can be considered as an element-wise function version of the
+    python keyword `in`. ``isin(a, b)`` is roughly equivalent to
+    ``np.array([item in b for item in a])`` if a is one-dimensional.
+    However, this idea fails if `b` is a set, or similar (non-sequence)
+    container:  As ``b`` is converted to an array, in those cases
+    ``asarray(b)`` is an object array containing b as its only element, 
+    rather than the expected array of contained values.
 
-    .. versionadded:: 1.4.0
+    .. versionadded:: 1.12.0
 
     Examples
     --------
-    >>> test = np.array([0, 1, 2, 5, 0])
-    >>> states = [0, 2]
-    >>> mask = np.in1d(test, states)
+    >>> test = np.array([[0, 2], [4, 6]])
+    >>> states = [1, 2, 4, 8]
+    >>> mask = np.isin(test, states)
     >>> mask
-    array([ True, False,  True, False,  True], dtype=bool)
+    array([[ False,  True],
+           [ True,  False]], dtype=bool)
     >>> test[mask]
-    array([0, 2, 0])
-    >>> mask = np.in1d(test, states, invert=True)
+    array([2, 4])
+    >>> mask = np.isin(test, states, invert=True)
     >>> mask
-    array([False,  True, False,  True, False], dtype=bool)
+    array([[ True, False],
+           [ False, True]], dtype=bool)
     >>> test[mask]
-    array([1, 5])
+    array([0, 6])
     """
-    # Ravel both arrays, behavior for the first array could be different
-    ar1 = np.asarray(ar1).ravel()
-    ar2 = np.asarray(ar2).ravel()
+    # Ravel both arrays, will reshape result later to match elements
```

Comment:
Even though `isin` should replace most uses of `in1d`, I think this code is actually easier to understand when a separate helper function is used for the 1D functionality, in the style of `unique`/`_unique1d` in this file. So I would keep the core logic in `in1d`.